relay-profiling/src/android.rs

```diff
@@ -217,6 +217,12 @@ pub fn parse_android_profile(
         profile.metadata.environment = environment.to_owned();
     }
 
+    if let Some(segment_id) = transaction_metadata.get("segment_id") {
+        if let Some(transaction_metadata) = profile.metadata.transaction.as_mut() {
+            transaction_metadata.segment_id = SpanId(segment_id.to_owned());
```

Comment:
Not part of this PR, but why does `transaction_metadata` exist as both a typed object and an untyped map on `ProfileMetadata`? Could we consolidate this somehow and make `extract_transaction_metadata` return a struct?

Reply:
@jjbayer quick answer is they're not exactly the same.

Here an example from a real profile:

```
    "transaction": {
        "active_thread_id": 134675695638272,
        "id": "89c91b9711e2479ab8ba56c88e2038f7",
        "name": "/api/0/organizations/{organization_slug}/events/",
        "trace_id": "84b8e259de564bc7b8622bb9f7fdb997"
    },
    "transaction_metadata": {
        "environment": "prod",
        "http.method": "GET",
        "release": "backend@03bd0a459082842c3a3963192123f75d91ab90d9",
        "transaction": "/api/0/organizations/{organization_slug}/events/",
        "transaction.end": "2024-03-05T14:42:39.082330942Z",
        "transaction.op": "http.server",
        "transaction.start": "2024-03-05T14:42:38.865197896Z",
        "transaction.status": "ok"
    },
```

Longer answer on why this was introduced to begin with, I think dates back to PR #1982

Reply:
OK! It would still be nice to type out both objects, but it does not have to be in this PR.